src/libcore/slice/mod.rs

```diff
@@ -53,6 +53,20 @@ mod sort;
 // Extension traits
 //
 
+/// Calculates the direction and split point of a one-sided range.
+///
+/// Helper for `take` and `take_mut`,
+#[inline]
+fn take_split_point(range: impl OneSidedRange<usize>) -> (bool, usize) {
+    match (range.start_bound(), range.end_bound()) {
+        (Bound::Unbounded, Bound::Excluded(i)) => (true, *i),
+        (Bound::Unbounded, Bound::Included(i)) => (true, *i + 1),
+        (Bound::Excluded(i), Bound::Unbounded) => (false, *i + 1),
```

Comment:
These two additions can overflow :/

For example, `[1, 2, 3].take(..=usize::MAX)` should return `None`, right? But instead, it will return `Some([])` due to overflow. I'm not quite sure what's the best solution is here. I guess you could use `check_add` and return an `Option`. If it's `None`, you also immediately return `None` in the methods below. At least as far as I can see, `take(..usize::MAX)` can never return `Some`: the `len` of the slice also has to be representable as `usize`, so the largest valid index is `usize::MAX - 1`.

Reply:
Good catch!